pkg/runtime/framework/plugins/coscheduling/indexer_test.go

```diff
@@ -0,0 +1,398 @@
+package coscheduling
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	jobsetv1alpha2 "sigs.k8s.io/jobset/api/jobset/v1alpha2"
+
+	trainer "github.com/kubeflow/trainer/pkg/apis/trainer/v1alpha1"
+	"github.com/kubeflow/trainer/pkg/constants"
+	utiltesting "github.com/kubeflow/trainer/pkg/util/testing"
+)
+
+func TestIndexTrainingRuntimeContainerRuntimeClass(t *testing.T) {
+	cases := map[string]struct {
+		obj  client.Object
+		want []string
+	}{
+
+		"object is not a TrainingRuntime": {
+			obj:  utiltesting.MakeTrainingRuntimeWrapper(metav1.NamespaceDefault, "test"),
```

Comment:
```suggestion
			obj:  utiltesting.MakeTrainingRuntimeWrapper(metav1.NamespaceDefault, "test").Obj(),
```

We should initialize it everywhere.